editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/actions/ui/SwitchCamelVersionWizard.java

```diff
@@ -39,18 +40,32 @@ public void addPages() {
 	@Override
 	public boolean performFinish() {
 		String selectedCamelVersion = page.getSelectedCamelVersion();
+		CamelVersionChecker camelVersionChecker = new CamelVersionChecker(selectedCamelVersion);
 		try {
-			getContainer().run(true, false, new SwitchCamelVersionRunnableWithProgress(this, selectedCamelVersion));
+			getContainer().run(true, true, camelVersionChecker);
 		} catch (InvocationTargetException e) {
 			ProjectTemplatesActivator.pluginLog().logError(e);
 		} catch (InterruptedException iex) {
 			ProjectTemplatesActivator.pluginLog().logError(iex);
 			Thread.currentThread().interrupt();
 		}
-		if(!hasValidCamelVersion) {
-			page.setErrorMessage(NLS.bind(Messages.invalidCamelVersionMessage, selectedCamelVersion));
+		while (!camelVersionChecker.isDone()) {
+			try {
+				Thread.sleep(100);
+			} catch (InterruptedException ex) {
+				Thread.currentThread().interrupt();
+			}
+		}
+		if (!camelVersionChecker.isCanceled()) {
+			hasValidCamelVersion = camelVersionChecker.isValid();
+
+			if(!hasValidCamelVersion) {
+				page.setErrorMessage(NLS.bind(Messages.invalidCamelVersionMessage, selectedCamelVersion));
+			}
+			return hasValidCamelVersion;
+		} else {
+			return false;
```

Comment:
can be moved out of the if-else clause, no?

Reply:
Do you mean writing it like that?

```
if (!camelVersionChecker.isCanceled()) {
	hasValidCamelVersion = camelVersionChecker.isValid();
	if(!hasValidCamelVersion) {
		page.setErrorMessage(NLS.bind(Messages.invalidCamelVersionMessage, selectedCamelVersion));
	}
		return hasValidCamelVersion;
}
return false;
```

I find it clearer to have the if/else expressed instead of using a default path.